--- a/src/common/ieee802_11_common.c
+++ b/src/common/ieee802_11_common.c
@@ -731,6 +731,20 @@
 
 	if (sec_channel > 1 || sec_channel < -1)
 		return NUM_HOSTAPD_MODES;
+	
+	/* HAMNET 13cm band, 2312..2377 MHz -> channels 1..14 */
+	if (freq >= 2312 && freq <= 2377) {
+		if ((freq - 2312) % 5)
+			return NUM_HOSTAPD_MODES;
+
+		if (vht)
+			return NUM_HOSTAPD_MODES;
+
+		*op_class = 81;
+		*channel = (freq - 2312) / 5 + 1;
+
+		return HOSTAPD_MODE_IEEE80211G;
+	}
 
 	if (freq >= 2412 && freq <= 2472) {
 		if ((freq - 2407) % 5)
