--- a/net/wireless/util.c	2026-06-14 12:45:47.072195149 +0200
+++ b/net/wireless/util.c	2026-06-14 12:46:28.506055276 +0200
@@ -122,6 +122,8 @@
 	struct ieee80211_supported_band *sband;
 	int i;
 
+    printk(KERN_DEBUG "HAMNET-DBG: get_channel freq=%d\n", freq);
+
 	for (band = 0; band < NUM_NL80211_BANDS; band++) {
 		sband = wiphy->bands[band];
 
@@ -134,6 +136,8 @@
 		}
 	}
 
+    printk(KERN_DEBUG "HAMNET-DBG: get_channel NO MATCH freq=%d\n", freq);
+	
 	return NULL;
 }
 EXPORT_SYMBOL(ieee80211_get_channel);
